### ISSUE TITLE
Support for old package.json license structure

### ISSUE THREAD

**Issue by 0xs** [xz64/license-webpack-plugin#5]
I know that according to the current package.json format specification the license info should always be a string. Unfortunately many packages still use the old notation of `license.type` shown [here](https://docs.npmjs.com/files/package.json#license). Since I should show these licenses too it would be great if there was support for listing them too.

**Comment by xz64** [comment]
I have pushed a change to the dev branch but haven't tested it. Please try the dev branch and see if that works out for you.

**Comment by grauwoelfchen** [comment]
Hi,

Thank you for nice tool. I like it!
I tried your **dev** branch. But I think that this line also should be changed for `moduleInfo.license`.

https://github.com/xz64/license-webpack-plugin/blob/a7607656420c69a23198eb8323b0d054865bbfda/index.js#L52

So, I've just tried to extract `license` before this pattern matching, and updated to support following old 2 styles.

```
license: {
  type: '...'
, url: '...'
}
```

\# Array but it contains only one license.
\# e.g. [respond.js](https://github.com/scottjehl/Respond/blob/master/package.json#L19)

```
licenses: [
  { 'type': '...'
  , 'url': '...'
  }
]
```

I'll attach a link of a small patch. could you please check this :)
https://gist.github.com/grauwoelfchen/d01899aa46c10f079dfdb8833a57df56

If `license` is also cached like packagejson, it might be better.

**Comment by xz64** [comment]
Thanks @grauwoelfchen . I've incorporated your patch and force pushed a new change to dev branch. Caching the license would be ideal, but it should be OK to not cache it for now. Please check out the dev branch and let me know if you have any other thoughts.

**Comment by grauwoelfchen** [comment]
Thank you @xz64 for quick response . You've also added improvements and tests :)
I will try dev branch!

If this plugin could support user to choose a license from packages which have muliple licenses in this old style licenses in `package.json`, it would be perfect to me!

**Comment by xz64** [comment]
@grauwoelfchen I have pushed a change to the dev branch which allows you to use a `licenseTypeOverrides` option. You can check the README to see how to use it. Please let me know if this works out for you.

**Comment by grauwoelfchen** [comment]
@xz64 Thank you su much! `licenseTypeOverrides` and `licenseTypeCache` look very nice :) I will try it.

**Comment by xz64** [comment]
Published this feature in v0.4.0.

**Comment by grauwoelfchen** [comment]
:tada:
